tests/validators/base.test.ts

```diff
@@ -0,0 +1,90 @@
+import { s } from '../../src';
+
+describe('BaseValidator', () => {
+	describe('optional', () => {
+		const optionalPredicate = s.string.optional;
+
+		test.each([undefined, 'hello'])('GIVEN %s THEN returns given value', (input) => {
+			expect(optionalPredicate.parse(input)).toEqual(input);
+		});
+	});
+
+	describe('nullable', () => {
+		const nullablePredicate = s.string.nullable;
+
+		test.each([null, 'Hello There'])('GIVEN %s THEN returns given value', (input) => {
+			expect<string | null>(nullablePredicate.parse(input)).toBe(input);
+		});
+	});
+
+	describe('nullish', () => {
+		const nullishPredicate = s.string.nullish;
+
+		test.each(['Hello There', undefined, null])('GIVEN %s THEN returns the given value', (input) => {
+			expect<string | undefined | null>(nullishPredicate.parse(input)).toBe(input);
+		});
+	});
+
+	describe('array', () => {
+		const arrayPredicate = s.number.array;
+
+		test('GIVEN an array of string THEN returns the given value', () => {
+			expect(arrayPredicate.parse([1, 2, 3])).toStrictEqual([1, 2, 3]);
+		});
+	});
+
+	describe('set', () => {
+		const setPredicate = s.number.set;
+
+		test('GIVEN a set of string THEN returns the given value', () => {
+			expect(setPredicate.parse(new Set([1, 2, 3]))).toStrictEqual(new Set([1, 2, 3]));
+		});
+	});
+
+	describe('or', () => {
+		const orPredicate = s.string.or(s.number);
+
+		test.each(['Hello There', 6])('GIVEN a string or number THEN returns a string', (input) => {
+			expect(orPredicate.parse(input)).toBe(input);
+		});
+	});
+
+	describe('transform', () => {
```

Comment:
We probably want to test transforms inside an union, e.g. what we have and `.or(s.number)`, to see that it only calls the callback when it's a string.

That'd be another test.

Reply:
I've added a test. Check if this is how u want?